irohad/ordering/impl/on_demand_ordering_gate.cpp

```diff
@@ -68,3 +69,40 @@ void OnDemandOrderingGate::setPcs(
   throw std::logic_error(
       "Method is deprecated. PCS observable should be set in ctor");
 }
+
+std::unique_ptr<shared_model::interface::Proposal>
+OnDemandOrderingGate::processProposalRequest(
+    const boost::optional<OnDemandOrderingService::ProposalType> &proposal)
+    const {
+  if (not proposal) {
+    return proposal_factory_->unsafeCreateProposal(
+      current_round_.block_round, current_round_.reject_round, {});
+  }
+  if (not *proposal) {
+    return nullptr;
+  }
+  return removeReplays(**proposal);
+}
+
+std::unique_ptr<shared_model::interface::Proposal>
+OnDemandOrderingGate::removeReplays(
+    const shared_model::interface::Proposal &proposal) const {
+  auto not_processed_txs = [this](const auto &tx) {
```

Comment:
I suggest renaming this function to something like `tx_is_not_processed`, because a) it currently looks very similar to the variable below, `unprocessed_txs`, which is the result of applying this filter, and b) it handles single tx, so no "txs"